File: backend/app/schemas/assistant.py
from datetime import datetime
from typing import Optional, List, Dict, Any
from pydantic import BaseModel, Field, model_validator

class AssistantMessageCreate(BaseModel):
    content: Optional[str] = None
    message: Optional[str] = None
    
    @model_validator(mode="after")
    def validate_message_fields(self):
        if not self.content and self.message:
            self.content = self.message
        elif not self.content and not self.message:
            raise ValueError("Either content or message must be provided")
        return self

class AssistantMessageResponse(BaseModel):
    id: int
    conversation_id: int
    sender_type: str
    content: str
    answer: Optional[str] = None
    response_metadata: Optional[Dict[str, Any]] = None
    created_at: datetime

    @model_validator(mode="after")
    def validate_answer(self):
        if not self.answer and self.content:
            self.answer = self.content
        return self

class AssistantConversationResponse(BaseModel):
    id: int
    learner_id: int
    title: Optional[str] = None
    created_at: datetime
    updated_at: datetime

class AIResponseContent(BaseModel):
    answer: str
    recommended_actions: Optional[List[str]] = None
    related_skill_ids: Optional[List[int]] = None
    related_activity_ids: Optional[List[int]] = None

File: frontend/src/api/assistant.ts
import api from './client';

export interface AssistantMessageCreate {
  content: string;
}

export interface AssistantMessageResponse {
  id: number;
  conversation_id: number;
  sender_type: 'USER' | 'ASSISTANT';
  content: string;
  response_metadata?: any;
  created_at: string;
}

export interface AssistantConversationResponse {
  id: number;
  learner_id: number;
  title?: string;
  created_at: string;
  updated_at: string;
}

export const assistantApi = {
  getConversations: async (): Promise<AssistantConversationResponse[]> => {
    const response = await api.get('/assistant/conversations');
    return response.data;
  },

  getMessages: async (conversationId: number): Promise<AssistantMessageResponse[]> => {
    const response = await api.get(`/assistant/conversations/${conversationId}/messages`);
    return response.data;
  },

  sendMessage: async (message: AssistantMessageCreate, conversationId?: number): Promise<AssistantMessageResponse> => {
    const url = conversationId ? `/assistant/chat?conversation_id=${conversationId}` : '/assistant/chat';
    const response = await api.post(url, message);
    return response.data;
  }
};
